File: examples/Grilli_rigid/create_mesh.py

import numpy as np


M = 500 #x divisions
N = 10 #y divisions
dx = 0.0125 #grid spacing


x0=0 # initial x coordinate
x = np.arange(x0, x0 + M*dx, dx) 


h0 = 0.33 #water depth
theta = 35*np.pi/180; #slope angle
x_s = 0.6; #slope finish


depth = np.empty((N, M))
depth[:, :] = h0
depth[:, x < x_s] = (-0.0901 + x * np.tan(theta))[x < x_s]
np.savetxt("depth.txt", depth, fmt="%.6f", delimiter=" ")

File: examples/Grilli_deformable_granular/create_mesh.m
M = 250;
N = 10;
dx = 0.025;

x = ((1:M)-0.5)*dx;


h0 = 0.33;
theta = 35*pi/180;
x_s = 0.6;
Ls = 0.9;
x_e = x_s + Ls;

depth = zeros(N,M);

for i = 1:M
    if x(i) < x_s
        depth(:,i) =-0.0901+ x(i)*tan(theta);
    else
        depth(:,i) = h0;
    end
end

dlmwrite('depth.txt', depth, 'delimiter', ' ', 'precision', 6);
